glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt

```diff
@@ -53,4 +103,361 @@ class GleanTest {
+        assertEquals(1, storedData.extra?.size)
+        assertEquals("value", storedData.extra?.getValue("test_key"))
+    }
+
+    @Ignore("Disabled because we're not triggering pings when going to background. See bug 1551159.")
+    @Test
+    fun `test sending of background pings`() {
+        val server = getMockWebServer()
+
+        val click = EventMetricType<NoExtraKeys>(
+            disabled = false,
+            category = "ui",
+            lifetime = Lifetime.Ping,
+            name = "click",
+            sendInPings = listOf("events")
+        )
+
+        resetGlean(getContextWithMockedInfo(), Glean.configuration.copy(
+            serverEndpoint = "http://" + server.hostName + ":" + server.port,
+            logPings = true
+        ))
+
+        // Fake calling the lifecycle observer.
+        val lifecycleRegistry = LifecycleRegistry(mock(LifecycleOwner::class.java))
+        val gleanLifecycleObserver = GleanLifecycleObserver()
+        lifecycleRegistry.addObserver(gleanLifecycleObserver)
+
+        try {
+            // Simulate the first foreground event after the application starts.
+            lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_RESUME)
+            click.record()
+
+            // Simulate going to background.
+            lifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_STOP)
+
+            // Trigger worker task to upload the pings in the background
+            triggerWorkManager()
+
+            val requests: MutableMap<String, String> = mutableMapOf()
+            for (i in 0..1) {
+                val request = server.takeRequest(20L, TimeUnit.SECONDS)
+                val docType = request.path.split("/")[3]
+                requests.set(docType, request.body.readUtf8())
+            }
+
+            val eventsJson = JSONObject(requests["events"])
+            checkPingSchema(eventsJson)
+            assertEquals("events", eventsJson.getJSONObject("ping_info")["ping_type"])
+            assertEquals(1, eventsJson.getJSONArray("events")!!.length())
+
+            val baselineJson = JSONObject(requests["baseline"])
+            assertEquals("baseline", baselineJson.getJSONObject("ping_info")["ping_type"])
+            checkPingSchema(baselineJson)
+
+            val baselineMetricsObject = baselineJson.getJSONObject("metrics")!!
+            val baselineStringMetrics = baselineMetricsObject.getJSONObject("string")!!
+            assertEquals(1, baselineStringMetrics.length())
+            assertNotNull(baselineStringMetrics.get("glean.baseline.locale"))
+
+            val baselineTimespanMetrics = baselineMetricsObject.getJSONObject("timespan")!!
+            assertEquals(1, baselineTimespanMetrics.length())
+            assertNotNull(baselineTimespanMetrics.get("glean.baseline.duration"))
+        } finally {
+            server.shutdown()
+            lifecycleRegistry.removeObserver(gleanLifecycleObserver)
+        }
+    }
+
+    @Ignore("This is currently failing")
```

Comment:
Do we have a bug for that?

Reply:
I filed [bug 1554081](https://bugzilla.mozilla.org/show_bug.cgi?id=1554081) for it.